tools/topology/topology1/development/sof-imx8-compr-pcm-wm8960.m4

```diff
@@ -0,0 +1,90 @@
+#
+# Topology with codec_adapter processing component for i.MX8QM/i.MX8QXP
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+include(`sai.m4')
+include(`pcm.m4')
+include(`buffer.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include DSP configuration
+include(`platform/imx/imx8.m4')
+
+
+# Post process setup config
+
+ #codec Post Process setup config
+#
+# Define the pipelines
+#
+# PCM0 ----> volume -----> SAI3 (wm8960)
+#
+
+DECLARE_SOF_RT_UUID("passthrough_codec", passthrough_uuid, 0x376b5e44, 0x9c82,
+		0x4ec2, 0xbc, 0x83, 0x10, 0xea, 0x10, 0x1a, 0xf8, 0x8f);
+
+define(`CA_UUID', passthrough_uuid)
+
+# Post process setup config for playback
+define(`CA_SETUP_CONTROLBYTES',
+``	bytes "0x53,0x4f,0x46,0x00,0x00,0x00,0x00,0x00,'
+`	0x00,0x00,0x00,0x00,0x00,0x10,0x00,0x03,'
+`	0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,'
+`	0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00"''
+)
+
+define(`CA_SETUP_CONTROLBYTES_MAX', 300)
+
+undefine(`DAI_PERIODS')
+define(`DAI_PERIODS', 8)
+
+dnl PIPELINE_PCM_ADD(pipeline,
```

Comment:
@LaurentiuM1234 one more thing. This will work fine with default DAI_PERIODS = 2. We needed the 'hack' with DAI_PERIODS = 8 for compressed streams. Which by the way we should look into that more closely as @plbossart suggested. 

I even think that after @ranj063 introduced local deep buffers also compress will work with DAI_PERIODS = 2.

Anyhow, for now I think we should use DAI_PERIODS = 2, and merge this.

Reply:
Fixed.